org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/NavigateToDefinitionHandlerTest.java

```diff
@@ -138,6 +141,32 @@ public void testJdkClasses() throws Exception {
 		testClass("org.apache.commons.lang3.stringutils", 6579, 20);
 	}
 
+	// this test should pass when starting with -javaagent:<lombok_jar> (-javagent:~/.m2/repository/org/projectlombok/lombok/1.18.20/lombok-1.18.20.jar)
+	// https://github.com/redhat-developer/vscode-java/issues/2805
+	@Test
+	public void testLombok() throws Exception {
+		boolean lombokDisabled = "true".equals(System.getProperty("jdt.ls.lombok.disabled"));
+		if (lombokDisabled) {
+			return;
+		}
+		importProjects("maven/mavenlombok");
+		IProject project = ResourcesPlugin.getWorkspace().getRoot().getProject("mavenlombok");
+		List<IMarker> markers = ResourceUtils.getErrorMarkers(project);
+		if (!markers.isEmpty()) {
+			// there isn't the lombok agent
+			return;
+		}
+		String uri = ClassFileUtil.getURI(project, "org.sample.Main");
+		TextDocumentIdentifier identifier = new TextDocumentIdentifier(uri);
+		List<? extends Location> locations = handler.definition(new TextDocumentPositionParams(identifier, new Position(5, 20)), monitor);
+		assertNotNull(locations);
+		assertEquals(1, locations.size());
+		assertEquals(6, locations.get(0).getRange().getStart().getLine());
```

Comment:
Just for completeness, I'd add an assertion here that the location URI is for the location of `org.sample.Test`.